app/models/beta/search/intercept_message.rb

```diff
@@ -1,12 +1,23 @@
 module Beta
   module Search
     class InterceptMessage
+      SECTION_REGEX = /(?<type>section)s? (?<roman_number>[XVI\d]{0,10})(?<terminator>[.,\s)])/i
       CHAPTER_REGEX = /(?<type>chapter)s? (?<optional>code )?(?<code>[0-9]{1,2})(?<terminator>[.,\s)])/i
       HEADING_REGEX = /(?<type>(?<!sub)heading)s? (?<optional>code )?(?<code>[0-9]{4})(?<terminator>[.,\s)])/i
       SUBHEADING_REGEX = /(?<type>subheading)s? (?<optional>code )?(?<code>[0-9]{6,8})(?<terminator>[.,\s)])/i
       COMMODITY_REGEX = /(?<type>commodity|commodities) (?<optional>code )?(?<code>[0-9]{10})(?<terminator>[.,\s)])/i
 
       GOODS_NOMENCLATURE_LINK_TRANSFORMERS = {
+        SECTION_REGEX => lambda do |matched_text|
+          match = matched_text.match(SECTION_REGEX)
+
+          section_id = RomanNumerals::Converter.to_decimal(match[:roman_number])
+
+          roman_section_id = RomanNumerals::Converter.to_roman(section_id)
```

Comment:
I guess you can pull this from the match[:roman_number] rather than converting again?

Reply:
(it is called just `code` now).
unfortunatelly, if the code is decimal, we cannot reuse `match[:code]`, because we want so show all the section links in Roman

Reply:
I don't think that's true. I think I went with not transforming the way the text for the link appears in the other examples basically to preserve the what the user wrote in their spreadsheet.